src/factories/ConfigurationFactory.js

```diff
@@ -0,0 +1,86 @@
+var uri = require("URIjs");
+
+export function ConfigurationFactory(sourceType, rawConfiguration) {
```

Comment:
I'll work on breaking it up. Would a class that has functions for each parsing operation be a good first step?